torch/csrc/jit/frontend/tracer.cpp

```diff
@@ -153,7 +153,19 @@ Value* TracingState::getValue(const IValue& var) {
       auto& value_map = env_stack.at(env_stack.size() - 1 - i);
       auto it = value_map.find(var);
       if (it == value_map.end()) {
-        continue;
+        // If we can't find corresponding TensorImpl based on ptr hash, perform search by
+        // trying to find a tensor based on matching storage alias and metadata...
+        it = value_map.begin();
```

Comment:
to limit the number of affected scenarios, is it possible to only take this path if the tensor requires grad? or are there cases where we have saved tensors that don't require grad?

Reply:
Good point @davidberard98! Let me take a look as I'm pretty sure we should be able to isolate it.

Reply:
Isolated the linear search only if tensor requires a gradient (https://github.com/pytorch/pytorch/pull/134879/commits/686110b0ecf750ad329a6ef3ff465aa846dc427e)